lib/lambda_cache.ex

```diff
@@ -3,24 +3,24 @@ defmodule LambdaCache do
   Zero-arity function polling cache.
   """
 
-  @behaviour refresh :: any()
-  @behaviour interval :: non_neg_integer()
+  @callback refresh :: any()
+  @callback interval :: non_neg_integer()
 
   defmacro __using__(options) do
     quote location: :keep do
       @behaviour LambdaCache
 
-      def child_spec(args) do
+      use GenServer
+
+      def child_spec(args \\ nil) do
         %{
           id: __MODULE__,
           start: {__MODULE__, :start_link, [args]}
         }
       end
 
-      use GenServer
-
       def start_link(args) do
-        GenServer.start_link(__MODULE__, args, unquote(options))
+        GenServer.start_link(__MODULE__, args, Keyword.merge([name: __MODULE__], unquote(options)))
```

Comment:
That's not a good default, this means every cache would be a singleton which would remove possibility say of dynamically creating caches at runtime.